core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java

```diff
@@ -394,4 +413,54 @@ private CloudSqlInstance apiFetcher(ConnectionConfig config) {
         localKeyPair,
         RateLimiter.create(1.0 / 30.0)); // 1 refresh attempt every 30 seconds
   }
+
+  ConnectionConfig resolveConfig(ConnectionConfig config) {
+    if (config.getNamedConnection() != null) {
+      ConnectionConfig namedConfig = namedConfigs.get(config.getNamedConnection());
+      if (namedConfig == null) {
+        throw new IllegalArgumentException(
+            "No named connection named " + config.getNamedConnection());
+      }
+      return namedConfig;
+    }
+    return config;
+  }
+
+  @Override
+  public void register(String name, ConnectionConfig config) {
+    ConnectionConfig oldConfig = this.namedConfigs.putIfAbsent(name, config);
```

Comment:
We should probably throw an exception if two registrations occur for the same name.

Reply:
Fixed.